src-tauri/src/main.rs

```diff
@@ -36,15 +39,27 @@ fn plot_progress_tracker() -> usize {
 async fn farming(path: String, reward_address: String, plot_size: u64) {
```

Comment:
Btw, does tauri support `Option<String>` and `PathBuf` for arguments? It would be handy to use those here

Reply:
unfortunately no :/ 
All the parameters and return types for `tauri command` functions need to be primitive types, since these are communicating with frontend.

Reply:
I believe `Option<String>` at least should be? There is `null` in js

Reply:
Maybe, but I think it serves no purpose for desktop app. Since there **always** will be `reward_address` and `path`. 

The reason for mandatory reward address: 
the first case: if the user decides to import reward address -> user enters the reward address, and frontend sends the reward address to backend.
the second case: user does not import reward address -> because of multi-replica, there has to be a ROOT-IDENTITY for reward address. We choose to generate this new identity on the frontend, for unified workflow (frontend sending the reward address to backend). 

In summary, there is no path in our workflow to *NOT* send a reward address to backend.

Reply:
Ok, should we delete branch there we check for `length == 0` on reward address?

Reply:
Missed changes in latter commits. It is no longer needed.